rust/transaction-filter/src/filter_operator.rs

```diff
@@ -0,0 +1,336 @@
+// use crate::traits::Filterable;
+use crate::{
+    filters::{
+        EventFilter, TransactionRootFilter, UserTransactionRequestFilter, WriteSetChangeFilter,
+    },
+    traits::Filterable,
+};
+use anyhow::Error;
+use aptos_protos::transaction::v1::{transaction::TxnData, Transaction};
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+/// These are filters we would expect to be exposed via API
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+#[serde(tag = "type")]
+pub enum APIFilter {
+    TransactionRootFilter(TransactionRootFilter),
+    UserTransactionRequestFilter(UserTransactionRequestFilter),
+    EventFilter(EventFilter),
+    WriteSetChangeFilter(WriteSetChangeFilter),
+}
+
+impl Filterable<Transaction> for APIFilter {
```

Comment:
You could use enum_dispatch / ambassador to help eliminate some of the boilerplate here.

Reply:
enum dispatchhhhh